oidc-exchange.py

```diff
@@ -140,81 +140,81 @@ def assert_successful_audience_call(resp: requests.Response, domain: str):
+                'audience retrieval failed: repository at '
+                f'{domain} responded with unexpected {other}: {status.phrase}',
             )
 
 
 def render_claims(token: str) -> str:
-    _, payload, _ = token.split(".", 2)
+    _, payload, _ = token.split('.', 2)
 
     # urlsafe_b64decode needs padding; JWT payloads don't contain any.
-    payload += "=" * (4 - (len(payload) % 4))
+    payload += '=' * (4 - (len(payload) % 4))
     claims = json.loads(base64.urlsafe_b64decode(payload))
 
     def _get(name: str) -> str:  # noqa: WPS430
-        return claims.get(name, "MISSING")
+        return claims.get(name, 'MISSING')
 
     return _RENDERED_CLAIMS.format(
-        sub=_get("sub"),
-        repository=_get("repository"),
-        repository_owner=_get("repository_owner"),
-        repository_owner_id=_get("repository_owner_id"),
-        job_workflow_ref=_get("job_workflow_ref"),
-        ref=_get("ref"),
+        sub=_get('sub'),
+        repository=_get('repository'),
+        repository_owner=_get('repository_owner'),
+        repository_owner_id=_get('repository_owner_id'),
+        job_workflow_ref=_get('job_workflow_ref'),
+        ref=_get('ref'),
     )
 
 
 def event_is_third_party_pr() -> bool:
     # Non-`pull_request` events cannot be from third-party PRs.
-    if os.getenv("GITHUB_EVENT_NAME") != "pull_request":
+    if os.getenv('GITHUB_EVENT_NAME') != 'pull_request':
         return False
 
-    event_path = os.getenv("GITHUB_EVENT_PATH")
+    event_path = os.getenv('GITHUB_EVENT_PATH')
     if not event_path:
         # No GITHUB_EVENT_PATH indicates a weird GitHub or runner bug.
-        debug("unexpected: no GITHUB_EVENT_PATH to check")
+        debug('unexpected: no GITHUB_EVENT_PATH to check')
         return False
 
     try:
         event = json.loads(Path(event_path).read_bytes())
     except json.JSONDecodeError:
-        debug("unexpected: GITHUB_EVENT_PATH does not contain valid JSON")
+        debug('unexpected: GITHUB_EVENT_PATH does not contain valid JSON')
         return False
 
     try:
-        return event["pull_request"]["head"]["repo"]["fork"]
+        return event['pull_request']['head']['repo']['fork']
     except KeyError:
         return False
 
 
-repository_url = get_normalized_input("repository-url")
+repository_url = get_normalized_input('repository-url')
 repository_domain = urlparse(repository_url).netloc
-token_exchange_url = f"https://{repository_domain}/_/oidc/mint-token"
+token_exchange_url = f'https://{repository_domain}/_/oidc/mint-token'
 
 # Indices are expected to support `https://{domain}/_/oidc/audience`,
 # which tells OIDC exchange clients which audience to use.
-audience_url = f"https://{repository_domain}/_/oidc/audience"
-audience_resp = requests.get(audience_url)
+audience_url = f'https://{repository_domain}/_/oidc/audience'
+audience_resp = requests.get(audience_url, timeout=5)  # S113 wants a timeout
```

Comment:
@woodruffw I've set a timeout of 5 here and in the following invocation to please the linter. Should we set a lower value or split connect/read?

Reply:
Hmm, that timeout seems fine to me (assuming that's seconds, which I think it is). IMO it _could_ be lower but I suspect the 99.99% case will fall well inside this (both for normal and outage responses).

Reply:
Alright, I'll let it be.